### PR TITLE
update to support body

### LINKED ISSUE #49: Request body not being sent to user container
The request body is not being sent to the user container.

### PR DESCRIPTION
**Pull request by beemarie**
# Changes
-  Fix lack of request body to user container
-  Add test to check for request body

/kind bug

Fixes #49

### CODE REVIEW

File: cmd/consumer/main.go [hunk 1]
```diff
@@ -47,7 +48,8 @@ func consumeEvent(event cloudevents.Event) error {
 
 	// client for sending request
 	client := &http.Client{}
-	req, err := http.NewRequest(data.ReqMethod, data.ReqURL, nil)
+	bodyReader := strings.NewReader(data.ReqBody)
```
Comment (vagababov) [inline cmd/consumer/main.go:51]: why bother with the variable?

File: cmd/consumer/main_test.go [hunk 2]
```diff
@@ -43,11 +44,20 @@ func TestConsumeEvent(t *testing.T) {
...
 			t.Errorf("Expected 'POST' OR 'GET' request, got '%s'", r.Method)
 		}
+		if r.Method == "POST" {
+			b, _ := ioutil.ReadAll(r.Body)
+			bodyString := string(b)
+			expectedBodyString := "{\"body\":\"test body\"}"
+			if bodyString != expectedBodyString {
+				t.Errorf("Expected body with POST request to match %s", expectedBodyString)
+			}
+		}
```
Comment (vagababov) [inline cmd/consumer/main_test.go:54]:
```suggestion
		switch r.Method {
		case http.MethodPost:
		...
		case http.MethodGet:
		default:
		// log error
		}
```

File: cmd/consumer/main_test.go [hunk 3]
```diff
@@ -77,7 +87,9 @@ func TestConsumeEvent(t *testing.T) {
 			data.ID = "123"
 			data.ReqURL = test.reqURL
 			data.ReqMethod = test.method
-
+			if data.ReqMethod == "POST" {
```
Comment (vagababov) [inline cmd/consumer/main_test.go:90]:
```suggestion
			if data.ReqMethod == http.MethodPost {
```

File: cmd/consumer/main_test.go [hunk 4]
```diff
@@ -77,7 +87,9 @@ func TestConsumeEvent(t *testing.T) {
...
+				data.ReqBody = "{\"body\":\"test body\"}"
```
Comment (vagababov) [inline cmd/consumer/main_test.go:91]:
```suggestion
				data.ReqBody = `{"body":"test body"}`
```

File: cmd/consumer/main_test.go (hunk 3 above)
Comment (beemarie) [reply cmd/consumer/main_test.go:91]: this was also in a number of other places -- fixed, thanks.

File: cmd/consumer/main_test.go [hunk 5]
```diff
@@ -40,14 +41,24 @@ func TestConsumeEvent(t *testing.T) {
 	myEvent.SetID("123")
 
 	testserver := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if r.Method != "GET" && r.Method != "POST" {
-			t.Errorf("Expected 'POST' OR 'GET' request, got '%s'", r.Method)
+		switch r.Method {
+		case http.MethodPost:
+			b, _ := ioutil.ReadAll(r.Body)
+			bodyString := string(b)
+			expectedBodyString := "{\"body\":\"test body\"}"
```
Comment (vagababov) [inline cmd/consumer/main_test.go:48]:
```suggestion
			const expectedBodyString = `{"body":"test body"}`
```

Comment (beemarie) [reply cmd/consumer/main_test.go]: ah, sorry - thanks!

**Review by maximilien — COMMENTED** [review]
/lgtm

**Pull request merged**